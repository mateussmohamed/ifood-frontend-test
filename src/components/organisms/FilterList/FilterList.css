.container-filter {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: row;
}

.filter-list {
  margin-bottom: 20px;
}

.filter-list-items {
  display: flex;
  flex-direction: row;
  margin: 8px 0;
}


@media screen and (max-width: 768px) {
  .container-filter {
    flex-direction: column;
    justify-content: center;
  }
  .filter-list-items {
    flex-wrap: wrap;
    justify-content: center;
  }
  .filter-button {
    margin: 5px;
  }
}

@media screen and (max-width: 480px) {
}
