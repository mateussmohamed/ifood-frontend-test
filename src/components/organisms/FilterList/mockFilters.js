export const localeFilters = {
  id: 'locale',
  name: 'Locale',
  values: [
    {
      value: 'en_AU',
      name: 'en_AU'
    },
    {
      value: 'de_DE',
      name: 'de_DE '
    },
    {
      value: 'pt_BR',
      name: 'pt_BR'
    },
    {
      value: 'fr_FR',
      name: 'fr_FR'
    },
    {
      value: 'en_US',
      name: 'en_US'
    },
    {
      value: 'es_AR',
      name: 'es_AR'
    }
  ]
};

export const countryFilters = {
  id: 'country',
  name: 'País',
  values: [
    {
      value: 'AU',
      name: 'Australia'
    },
    {
      value: 'DE',
      name: 'Alemanha'
    },
    {
      value: 'BR',
      name: 'Brasil'
    },
    {
      value: 'PT',
      name: 'Portugal'
    },
    {
      value: 'US',
      name: 'EUA'
    },
    {
      value: 'RU',
      name: 'Rússia'
    }
  ]
};

export const timestampFilters = {
  id: 'timestamp',
  name: 'Data e Horário',
  validation: {
    primitiveType: 'STRING',
    entityType: 'DATE_TIME',
    pattern: 'yyyy-MM-ddTHH:mm:ss'
  }
};
export const limitFilters = {
  id: 'limit',
  name: 'Quantidade',
  validation: {
    primitiveType: 'INTEGER',
    min: 1,
    max: 50
  }
};

export const offsetFilters = {
  id: 'offset',
  name: 'Página',
  validation: {
    primitiveType: 'INTEGER'
  }
};
