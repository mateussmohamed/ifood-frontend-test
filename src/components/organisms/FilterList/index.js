import React from 'react';
import PropTypes from 'prop-types';
import FilterTitle from '../../atoms/FilterTitle';
import FilterButton from '../../molecules/FilterButton';

import { parseFilter } from '../../../utils';

import './FilterList.css';

const isActive = (actives, filter, val) => (actives !== undefined && val !== undefined) && actives[filter] === val;

const FilterList = ({ data, action, actives }) => (
  data.map(filter => filter.values &&
    (
      <section className="filter-list" key={filter.name} >
        <FilterTitle text={filter.name} />
        <div className="filter-list-items">
          {
            filter.values.map((val) => (
              <FilterButton
                key={val.name}
                text={val.name}
                active={isActive(actives, filter.id, val.value) || false}
                action={action(parseFilter(filter.id, val.value))}
              />
            ))
          }
        </div>
      </section>
    )
  )
);


FilterList.defaultProps = {
  data: [],
};

FilterList.propTypes = {
  data: PropTypes.arrayOf(PropTypes.object),
  action: PropTypes.func.isRequired,
  actives: PropTypes.objectOf(PropTypes.any)
};

export default FilterList;
