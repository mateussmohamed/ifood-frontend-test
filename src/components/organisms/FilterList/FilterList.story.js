import { storiesOf } from '@storybook/react';
import React from 'react';
import FilterList from './index';
import mockFilters from './mockFilters';

const stories = storiesOf('FilterList', module);

stories.add('default', () => (
  <section>
    <FilterList data={mockFilters} action={e => () => alert(e)} />
  </section>
));
