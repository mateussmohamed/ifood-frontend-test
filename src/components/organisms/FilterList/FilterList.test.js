import React from 'react';
import renderer from 'react-test-renderer';
import FilterList from './index';
import mockFilters from './mockFilters';

it('Should FilterList match to snapshot', () => {
  const component = (
    <FilterList
      data={mockFilters}
      action={e => () => console.log(e)}
    />
  );
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});
