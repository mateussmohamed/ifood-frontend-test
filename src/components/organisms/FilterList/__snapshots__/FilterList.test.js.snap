// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should FilterList match to snapshot 1`] = `
Array [
  <section
    className="filter-list"
>
    <div
        className="filter-title"
    >
        Locale
    </div>
    <div
        className="filter-list-items"
    >
        <button
            className="filter-button "
            onClick={[Function]}
        >
            <div
                className="filter-title"
            >
                en_AU
            </div>
        </button>
        <button
            className="filter-button "
            onClick={[Function]}
        >
            <div
                className="filter-title"
            >
                de_DE 
            </div>
        </button>
        <button
            className="filter-button "
            onClick={[Function]}
        >
            <div
                className="filter-title"
            >
                pt_BR
            </div>
        </button>
        <button
            className="filter-button "
            onClick={[Function]}
        >
            <div
                className="filter-title"
            >
                fr_FR
            </div>
        </button>
        <button
            className="filter-button "
            onClick={[Function]}
        >
            <div
                className="filter-title"
            >
                en_US
            </div>
        </button>
        <button
            className="filter-button "
            onClick={[Function]}
        >
            <div
                className="filter-title"
            >
                es_AR
            </div>
        </button>
    </div>
</section>,
  <section
    className="filter-list"
>
    <div
        className="filter-title"
    >
        País
    </div>
    <div
        className="filter-list-items"
    >
        <button
            className="filter-button "
            onClick={[Function]}
        >
            <div
                className="filter-title"
            >
                Australia
            </div>
        </button>
        <button
            className="filter-button "
            onClick={[Function]}
        >
            <div
                className="filter-title"
            >
                Alemanhã
            </div>
        </button>
        <button
            className="filter-button "
            onClick={[Function]}
        >
            <div
                className="filter-title"
            >
                Brasil
            </div>
        </button>
        <button
            className="filter-button "
            onClick={[Function]}
        >
            <div
                className="filter-title"
            >
                Portugal
            </div>
        </button>
        <button
            className="filter-button "
            onClick={[Function]}
        >
            <div
                className="filter-title"
            >
                EUA
            </div>
        </button>
        <button
            className="filter-button "
            onClick={[Function]}
        >
            <div
                className="filter-title"
            >
                Rússia
            </div>
        </button>
    </div>
</section>,
]
`;
